website/src/app/api/deposit/estimate-claim/route.ts

```diff
@@ -0,0 +1,57 @@
+import {NextRequest, NextResponse} from 'next/server';
+import {Calldata, stark} from 'starknet';
+
+import {ESCROW_ADDRESSES} from '@/constants/contracts';
+import {Entrypoint} from '@/constants/misc';
+import {account} from '@/services/account';
+import {provider} from '@/services/provider';
+import {ErrorCode} from '@/utils/errors';
+import {HTTPStatus} from '@/utils/http';
+import {ClaimSchema} from '@/utils/validation';
+
+import {getClaimCallData} from '../calldata';
+
+export async function POST(request: NextRequest) {
+  const requestBody = await request.json();
+
+  const body = ClaimSchema.safeParse(requestBody);
+  if (!body.success) {
+    return NextResponse.json(
+      {code: ErrorCode.BAD_REQUEST, error: body.error},
+      {status: HTTPStatus.BadRequest},
+    );
+  }
+
+  let claimCallData: Calldata;
+  try {
+    const {calldata} = await getClaimCallData(body.data);
+    claimCallData = calldata;
+  } catch (error) {
+    if (error instanceof Error) {
+      return NextResponse.json({code: error.message}, {status: HTTPStatus.BadRequest});
+    }
+
+    throw error;
+  }
+
+  try {
+    const result = await account.estimateInvokeFee([

```

Comment:
In the end does it need a signature or not?

Reply:
For infura it does, for others it doesn't. However starknet.js signs the transaction as a real invocation, not as an estimation, so opting-out and not using signature.